mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/impl/HighLevelMqttSource.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.mqtt.streaming.impl
+
+import akka.actor.ActorSystem
+import akka.annotation.InternalApi
+import akka.dispatch.ExecutionContexts
+import akka.stream.alpakka.mqtt.streaming._
+import akka.stream.alpakka.mqtt.streaming.scaladsl._
+import akka.stream.scaladsl.{BroadcastHub, Flow, Keep, RestartFlow, Source, SourceQueueWithComplete}
+import akka.stream.{Materializer, OverflowStrategy}
+import akka.{Done, NotUsed}
+
+import scala.collection.immutable
+import scala.concurrent.{Future, Promise}
+
+/**
+ * Internal API
+ */
+@InternalApi
+private[streaming] object HighLevelMqttSource {
+
+  def atMostOnce(
+      mqttClientSession: MqttClientSession,
+      transportSettings: MqttTransportSettings,
+      restartSettings: MqttRestartSettings,
+      connectionSettings: MqttConnectionSettings,
+      subscriptions: MqttSubscribe
+  ): Source[Publish, Future[immutable.Seq[(String, ControlPacketFlags)]]] = {
+    type Out = Publish
+
+    val sendAcknowledge: SourceQueueWithComplete[Command[Nothing]] => PartialFunction[Event[Nothing], Out] =
```

Comment:
Shouldn't you type param the `[Nothing]`?

Reply:
This is "just internal", we can change it when it becomes necessary.

Reply:
OK.